sdk/azcore/runtime/pipeline.go

```diff
@@ -11,26 +11,34 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 )
 
+// SDKOptions contains Pipeline options for SDK developers
+type SDKOptions struct {
+	AllowedHeaders, AllowedQueryParameters []string
```

Comment:
Why are these duplicated from `LogOptions`, what's the general pattern here?

Reply:
This is to allow an SDK developer to augment the `ClientOptions` provided by the application without mutating it. The SDK developer could do that by cloning the application's `ClientOptions` and augmenting the clone--we could make that easier with a helper--but I believe @JeffreyRichter wants the flexibility to grow this new struct without having to extend `ClientOptions` as well.

Reply:
Yes, plus we never want to mutate the csutomer's ClientOptions because they use it to create multiple clients.

Reply:
I'm onboard with having a struct here so we can expand the `NewPipeline` funcs, but we should change the name to match our other patterns.

I don't think we need to duplicate any options fields here though to achieve the goal of not mutating caller-provided state.